src/panels/config/automation/trigger/types/ha-automation-trigger-webhook.ts

```diff
@@ -1,39 +1,141 @@
+    super.disconnectedCallback();
+    if (this._unsub) {
+      this._unsub();
+    }
+  }
+
+  private _generateWebhookId(): string {
+    // The webhook_id should be treated like a password. Generate a default
+    // value that would be hard for someone to guess. This generates a
+    // 144-bit random value. The output is a 24 character url-safe string.
+    const randomBytes = crypto.getRandomValues(new Uint8Array(18));
+    const base64Str = btoa(String.fromCharCode(...randomBytes));
+    const urlSafeId = base64Str.replace(/\+/g, "-").replace(/\//g, "_");
+
+    // Include the automation name to give the user context about what the
+    // webhook_id is used for.
+    const urlSafeAlias = slugify(this._config?.alias || "", "-");
+
+    return `${urlSafeAlias}-${urlSafeId}`;
+  }
+
+  public willUpdate(changedProperties: PropertyValues) {
+    super.willUpdate(changedProperties);
+    if (changedProperties.has("trigger")) {
+      if (this.trigger.webhook_id === DEFAULT_WEBHOOK_ID) {
+        this.trigger.webhook_id = this._generateWebhookId();
+      }
+    }
   }
 
   protected render() {
     const { webhook_id: webhookId } = this.trigger;
+
     return html`
-      <paper-input
+      <ha-textfield
         .label=${this.hass.localize(
           "ui.panel.config.automation.editor.triggers.type.webhook.webhook_id"
         )}
-        name="webhook_id"
+        .helper=${this.hass.localize(
+          "ui.panel.config.automation.editor.triggers.type.webhook.webhook_id_helper"
+        )}
+        .iconTrailing=${true}
         .value=${webhookId}
-        @value-changed=${this._valueChanged}
-      ></paper-input>
+        @input=${this._webhookIdChanged}
+      >
+        <ha-icon-button
+          @click=${this._copyUrl}
+          slot="trailingIcon"
+          .label=${this.hass.localize(
+            "ui.panel.config.automation.editor.triggers.type.webhook.copy_url"
+          )}
+          .path=${mdiContentCopy}
+        ></ha-icon-button>
+      </ha-textfield>
     `;
   }
 
-  private _valueChanged(ev: CustomEvent): void {
-    handleChangeEvent(this, ev);
+  private _webhookIdChanged(ev: CustomEvent): void {
+    ev.stopPropagation();
+    const newValue = (ev.currentTarget as HaTextField).value;
+    if (this.trigger.webhook_id === newValue) {
+      return;
+    }
+    const newTrigger = { ...this.trigger, webhook_id: newValue };
+    fireEvent(this, "value-changed", { value: newTrigger });
   }
+
+  private async _copyUrl(ev): Promise<void> {
+    const inputElement = ev.target.parentElement as HaTextField;
+    const url = this.hass.hassUrl(`/api/webhook/${inputElement.value}`);
+
+    await copyToClipboard(url);
+    showToast(this, {
+      message: this.hass.localize("ui.common.copied_clipboard"),
+    });
+  }
+
+  static styles = css`
+    ha-textfield {
+      display: block;
+      --mdc-text-field-fill-color: var(
+        --ha-card-background,
+        var(--card-background-color, white)
+      );
+    }
```

Comment:
```suggestion
    ha-textfield {
      display: block;
    }
```
We are updating our inputs to the new material standards, that includes the background on the input element.